src/path_calculator.rs

```diff
@@ -161,8 +176,9 @@ impl PathCalculator {
         if meeting_node == INVALID_NODE {
             return None;
         } else {
-            let node_ids = self.extract_nodes(graph, start, end, meeting_node);
-            return Some(ShortestPath::new(start, end, best_weight, node_ids));
+            let node_ids = self.extract_nodes(graph, end, meeting_node);
+            let chosen_start = node_ids[0];
+            return Some(ShortestPath::new(chosen_start, end, best_weight, node_ids));
```

Comment:
And should we not also include the start weight here? I guess it's a matter of convention. We could either think of the start weights as 'priorities' that can just be used to make it more/less likely for certain nodes to end up being the starting node of the shortest paths, or we could think of the start weights as actual weights that will be included in the returned path weight. I think I would prefer the second option, but maybe there is a reason you did not include it here?